File: docs/js/vegas-time.js
/** Las Vegas local time — Pacific Time (America/Los_Angeles), two hours behind Central. */
export const VEGAS_TIMEZONE = "America/Los_Angeles";

const VEGAS_LOCALE = "en-US";

function parseDate(isoOrDate) {
  if (isoOrDate instanceof Date) return isoOrDate;
  const d = new Date(isoOrDate);
  return Number.isNaN(d.getTime()) ? null : d;
}

/** Full Las Vegas date/time in 24-hour format with PT suffix. */
export function formatVegasDateTime(isoOrDate) {
  const d = parseDate(isoOrDate);
  if (!d) return "unknown";
  const formatted = d.toLocaleString(VEGAS_LOCALE, {
    timeZone: VEGAS_TIMEZONE,
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    second: "2-digit",
    hour12: false,
  });
  return `${formatted} PT`;
}

/** Compact Las Vegas date/time for save slots and ledgers. */
export function formatVegasDateTimeShort(isoOrDate) {
  const d = parseDate(isoOrDate);
  if (!d) return "unknown";
  const formatted = d.toLocaleString(VEGAS_LOCALE, {
    timeZone: VEGAS_TIMEZONE,
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    hour12: false,
  });
  return `${formatted} PT`;
}

/** Las Vegas wall-clock time in 24-hour format (HH:MM:SS). */
export function formatVegasTime(isoOrDate) {
  const d = parseDate(isoOrDate ?? new Date());
  if (!d) return "unknown";
  return d.toLocaleTimeString(VEGAS_LOCALE, {
    timeZone: VEGAS_TIMEZONE,
    hour: "2-digit",
    minute: "2-digit",
    second: "2-digit",
    hour12: false,
  });
}

/** Current Las Vegas wall-clock time. */
export function formatVegasNow() {
  return formatVegasTime(new Date());
}

/** Label for live clock displays in the casino UI. */
export function formatVegasClockLabel(isoOrDate) {
  return `Las Vegas: ${formatVegasTime(isoOrDate ?? new Date())} PT`;
}

/** Guest book / conversation timestamps. */
export function formatVegasSignedAt(isoOrDate) {
  const d = parseDate(isoOrDate);
  if (!d) return String(isoOrDate ?? "unknown");
  const formatted = d.toLocaleString(VEGAS_LOCALE, {
    timeZone: VEGAS_TIMEZONE,
    year: "numeric",
    month: "short",
    day: "numeric",
    hour: "2-digit",
    minute: "2-digit",
    hour12: false,
  });
  return `${formatted} PT`;
}

File: mandalay_bay/vegas_time.py
"""Las Vegas local time — Pacific Time (America/Los_Angeles), two hours behind Central."""

from __future__ import annotations

from datetime import datetime, timezone
from zoneinfo import ZoneInfo

VEGAS_TZ = ZoneInfo("America/Los_Angeles")


def to_vegas(dt: datetime) -> datetime:
    if dt.tzinfo is None:
        dt = dt.replace(tzinfo=timezone.utc)
    return dt.astimezone(VEGAS_TZ)


def format_vegas_datetime(dt: datetime | None) -> str:
    if dt is None:
        return "never"
    return to_vegas(dt).strftime("%Y-%m-%d %H:%M PT")


def format_vegas_time(dt: datetime) -> str:
    return to_vegas(dt).strftime("%H:%M:%S")


def format_vegas_now() -> str:
    return format_vegas_time(datetime.now(timezone.utc))


def format_vegas_clock_label() -> str:
    return f"Las Vegas: {format_vegas_now()} PT"


def format_vegas_signed_at(iso: str) -> str:
    dt = datetime.fromisoformat(iso.replace("Z", "+00:00"))
    return to_vegas(dt).strftime("%b %d, %Y %H:%M PT")
